src/shogun/regression/LeastAngleRegression.cpp

```diff
@@ -174,10 +180,13 @@ bool CLeastAngleRegression::train_machine(CFeatures* data)
 	int32_t nloop=0;
 	while (m_num_active < max_active_allowed && max_corr > CMath::MACHINE_EPSILON && !stop_cond)
 	{
+		// c_k = x_t*r_k
 		// corr = X' * (y-mu) = - X'*mu + Xy
 		copy(Xy.begin(), Xy.end(), corr.begin());
-		cblas_dgemv(CblasColMajor, CblasTrans, n_vec, n_fea, -1,
-			X.matrix, n_vec, &mu[0], 1, 1, &corr[0], 1);
+		Map<VectorXd> map_corr(&corr[0], n_fea);
+		Map<VectorXd> map_mu(&mu[0], n_vec);
+		
+		map_corr=map_Xy-(map_X.transpose()*map_mu);
```

Comment:
but that lapack thing wasn't even doing this... it was that manual nested loop in the beginning of `train_machine` that created this :P